src/mumble/GlobalShortcut_win.cpp

```diff
@@ -216,6 +216,60 @@ bool GlobalShortcutWin::event(QEvent *event) {
 	return QObject::event(event);
 }
 
+void GlobalShortcutWin::injectKeyboardMessage(MSG *msg) {
+	// Only allow keyboard messages.
+	switch (msg->message) {
+		case WM_KEYDOWN:
+		case WM_KEYUP:
+		case WM_SYSKEYDOWN:
+		case WM_SYSKEYUP:
+			break;
+		default:
+			return;
+	}
+
+	DWORD scancode = (msg->lParam >> 16) & 0xff;
+	DWORD vkcode = msg->wParam;
+	bool extended = !!(msg->lParam & 0x01000000);
+	bool up = !!(msg->lParam & 0x80000000);
+
+	qApp->postEvent(this, new InjectKeyboardMessageEvent(scancode, vkcode, extended, !up));
+}
+
+void GlobalShortcutWin::injectMouseMessage(MSG *msg) {
+	bool down = false;
+	unsigned int btn = 0;
+
+	switch (msg->message) {
+		case WM_LBUTTONDOWN:
+			down = true;
+		case WM_LBUTTONUP:
```

Comment:
For things such as push-to-talk, it's important to know the exact state of the button. You enter talking state on push, you stop talking when you release. Makes sense?